docs/vfio-user.rst

```diff
@@ -1127,6 +1140,111 @@ client must write data on the same order and transction size as read.
+| Error        | 0/errno                |
++--------------+------------------------+
+| Region IO info  | Region IO fd info       |
++--------------+------------------------+
+
+This command message is sent by the client to the server to query for
+sub-regions of the given region that should be accessed via file descriptors.
+
+A client should hook up the returned file descriptors as requested; when
+accesses are made to these sub-regions, the file descriptor is directly
+notified/written (for example, by KVM), avoiding a round-trip through the client
+and a VFIO_USER_REGION_READ/WRITE call.
+
+
+Region IOFD info format
+^^^^^^^^^^^^^^^^^^^^^^^
+
++------------+--------+------------------------------+
+| Name       | Offset | Size                         |
++============+========+==============================+
+| argsz      | 16     | 4                            |
++------------+--------+------------------------------+
+| flags      | 20     | 4                            |
++------------+--------+------------------------------+
+|            | +-----+-----------------------------+ |
+|            | | Bit | Definition                  | |
+|            | +=====+=============================+ |
+|            | | 0   | FIXME
+|            | +-----+-----------------------------+ |
++------------+--------+------------------------------+
+| index      | 24     | 4                            |
++------------+--------+------------------------------+
+| nr_subregions     | 28     | number of subregions in array |
++------------+--------+------------------------------+
+| subregions | 32     | array of subregions          |
++------------+--------+------------------------------+
+
+* *index* is the index of memory region being queried, it is the only field
+  that is required to be set in the command message.
+* *argsz* is the size of the VFIO region IOFD info structure plus the
+  size of the subregion array (such that each array entry is defined by this
+  size, minus the offset, divided by nr_subregions)
+* nr_subregions is the number of entries in the subregion array
+
+The reply message will additionally include at least one file descriptor in the
+ancillary data. Note that more than one subregion may share the same file
+descriptor.
+
+VFIO user subregion info
+------------------------
+
+https://www.spinics.net/lists/kvm/msg208139.html
+https://patchwork.kernel.org/project/kvm/patch/1251028605-31977-23-git-send-email-avi@redhat.com/
+
+FIXME rewrite in table format
+
+struct vfio_user_subregion_info {
+    // offset within region (not GPA)
+    u64 offset;
+    // size of sub-region (perhaps zero depending on flags below)
+    u64 size;
+    // index into ancillary data fd array
+    int fd_index;
+    // ioeventfd | ioregionfd
+    u32 subregion_type;
+
+    // FIXME: if ioregionfd subsumes ioeventfd, we could potentially just have
+    // the one struct; ioregionfd-less systems could fall back to ioeventfd
+    // interfaces
+    union {
```

Comment:
I don't understand most of these fields, could you elaborate?

Reply:
I updated the links above so it's clearer where to look for that info